sql: add plpgsql parser scaffold

chengxiong-ruan opened the pull request:
code generation works but it does not parse anything yet
Will come back to this maybe next breather week for the sake of #91569

Epic: None

chengxiong-ruan committed:
sql: add plpgsql parser scaffold.
code generation works but it does not parse anything yet.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/91606)
<!-- Reviewable:end -->

chengxiong-ruan committed:
sql: declare section and exec sql

very first test to parse an empty function or function with bare
sql queries.

Release note: None

rharding6373 committed:
sql: Adds support for stmt_case

Does not yet support ELSE, sql statements in THEN body, or END CASE (it
uses an ENDCASE hack).

Release note: None

ZhouXing19 committed:
[jane] lexer.ReadSqlExpressionStr() and stmt_if

chengxiong-ruan committed:
sql: plpgsql decl_header and stmt_assert

Release note: None

chengxiong-ruan committed:
sql: stmt_call, stmt_close, stmt_dynaexec, stmt_exit

Release note: None

ZhouXing19 committed:
[jane] stmt_assign and stmt_getdiag

chengxiong-ruan committed:
sql: fix endif and endcase

ZhouXing19 committed:
[jane] complete stmt_if, elsif, else

<what was there before: Previously, ...>
<why it needed to change: This was inadequate because ...>
<what you did about it: To address this, this patch ...>

ZhouXing19 committed:
stmt_open

chengxiong-ruan commented:
closing this since the better refactored ones were merged.
